naga/src/proc/constant_evaluator.rs

```diff
@@ -781,7 +781,14 @@ impl<'a> ConstantEvaluator<'a> {
                 })
             }
             crate::MathFunction::Pow => self.math_pow(arg, arg1.unwrap(), span),
-            crate::MathFunction::Clamp => self.math_clamp(arg, arg1.unwrap(), arg2.unwrap(), span),
+            crate::MathFunction::Clamp => {
+                component_wise_scalar!(
+                    self,
+                    span,
+                    [arg, arg1.unwrap(), arg2.unwrap()],
+                    |e, low, high| { e.clamp(low, high).into() }
```

Comment:
We were previously erroring with `ConstantEvaluatorError::InvalidClamp` if `low > high` we should still do that since the spec requires it and rust's `clamp` fn will panic in this case.

Reply:
Good catch! I've addressed the `low > high` check with 88659b92e, but checking [`f32::clamp`](https://doc.rust-lang.org/std/primitive.f32.html#method.clamp), we also have a mismatch when `low` or `high` are NaN (which would panic on the Rust side). 🤔 I'm not sure what the expected output would be here; do you know?

Reply:
The spec says that NaNs and Infinities are not allowed in const contexts, the check below validates those out.

https://github.com/gfx-rs/wgpu/blob/4b82121501a61c2c2e11cb472d70ba54af3aa12d/naga/src/proc/constant_evaluator.rs#L1548-L1550